feat: include tooling removal methods

CompeyDev opened the pull request:
Adds a `rokit remove` command to remove installed tools, which works similar to `rokit add`.

filiptibell commented on lib/manifests/rokit.rs:
```diff
@@ -132,6 +132,27 @@ impl RokitManifest {
     }
 
+    /**
+        Removes a tool to the manifest.
```
```suggestion
        Removes a tool from the manifest.
```

filiptibell commented on lib/manifests/rokit.rs:
```diff
@@ -132,6 +132,27 @@ impl RokitManifest {
+    /**
+        Removes a tool to the manifest.
+
+        If the tool does not exist, this will return `false` and do nothing.
```
```suggestion
        If the tool did not exist, this will return `false` and do nothing.
```

filiptibell commented on src/cli/remove.rs:
```diff
@@ -0,0 +1,87 @@
+        if tool_cache.is_installed(&spec) {
+            pt.update_message("Uninstalling");
+            tool_storage.remove_tool_link(&self.alias).await?;
+            pt.task_completed();
+        } else {
+            pt.task_completed();
+            pt.task_completed();
+        }
```
```suggestion
        }
        pt.task_completed();
```

filiptibell commented on src/cli/remove.rs:
```diff
@@ -0,0 +1,87 @@
+
+use crate::util::CliProgressTracker;
+
+/// Adds a new tool to Rokit and installs it.
```
```suggestion
/// Removes a tool from Rokit.
```

filiptibell commented on src/cli/remove.rs:
```diff
@@ -0,0 +1,87 @@
+pub struct RemoveSubcommand {
+    /// The alias of the tool to remove.
+    pub alias: ToolAlias,
+    /// Remove this tool globally instead of adding
+    /// it to the nearest manifest file.
```
```suggestion
    /// Remove this tool globally instead of removing
    /// it from the nearest manifest file.
```

CompeyDev replied:
Oops, didn't edit the doc comments for when I copied the related functions to modify!

filiptibell commented on src/cli/mod.rs:
```diff
@@ -1,5 +1,6 @@
 use anyhow::{Context, Result};
 use clap::{ArgAction, CommandFactory, Parser};
+use remove::RemoveSubcommand;
```
I assume this was an auto import, so rust-analyzer put it up here, but all the imports for commands are grouped below so the new import should be there as well.

filiptibell reviewed (changes requested):
Went through the code and it looks good overall, and it looks like this is mostly copied from the `add` command, so I just left a few nits.
However, the reason the `remove` command was not implemented before Rokit's release is because there were some additional things to consider. They all stem from the fact that Rokit does not and can not know about all `rokit.toml` files on the system.

1. Should the `remove` command only remove tools from the single manifest, similar to `add`? If so, how do we communicate to a user that the tool may still exist on their system, just not in the chosen manifest?
2. If the command should remove tool binaries, which ones does it remove?
    - All of them, requiring a reinstall in Rokit projects that may still be using the tool
    - Only the one listed in the manifest, leaving remaining binaries, but the exact tool may still need a reinstall in other Rokit projects
3. If the command should remove tool *links*, how do we make sure this is actually correct? A user can have a tool such as `rojo = "Kampfkarren/selene@x.y.z"` if they so desire, and this would actually need the `rojo` link to still exist.

I dont have a good answer to these questions but it seems to me like https://github.com/rojo-rbx/rokit/pull/70 considered it to some extent (it removed the link only when all tool binaries were gone).
Maybe it would be good to have some combination of a `remove` + `uninstall` subcommand, where `uninstall` nukes the binaries while `remove` only deals with manifests.

ActualMasterOogway commented:
@filiptibell any plans on reviewing this PR again?

CompeyDev commented:
> @filiptibell any plans on reviewing this PR again?

I haven't yet addressed all the comments on this PR, and I do not think I will be able to anytime soon. 

The feature is pretty simple to implement, and I have enabled maintainer commits, so @filiptibell should be able to finalize and merge it.
